webapp/src/components/Quiz/Quiz.js

```diff
@@ -1,144 +1,254 @@
-      if (currentQuestionIndex < questions.length - 1) {
-        setCurrentQuestionIndex(prevIndex => prevIndex + 1)
-      } else {
-        setIsFinished(true)
-      }
-
-      // Reset state for the next question
-      setQuizData({})
-      setBtnDisabled(false)
-      setTimer(Date.now() + time * 1000)
-      setTimerIndex(prevTimerIndex => prevTimerIndex + 1)
-    }, 2000)
-
-    // Pause the timer
-    pauseTimer()
-  }
-
-  const tryAgainHandler = () => {
-    setCurrentQuestionIndex(0)
-    setQuizData({})
-    setIsFinished(false)
-    setScore(0)
-    setBtnDisabled(false)
-    setTimer(Date.now() + time * 1000)
-    setTimerIndex(prevState => {
-      return prevState + 1
-    })
-  }
+      if (currentQuestionIndex < questions.length - 1)
+        setCurrentQuestionIndex((prevIndex) => prevIndex + 1);
+      else setIsFinished(true);
+
+      setQuizData({});
+      setBtnDisabled(false);
+      setTimer(Date.now() + timerValue * 1000);
+      setTimerIndex((prevTimerIndex) => prevTimerIndex + 1);
+    }, 2000);
+
+    pauseTimer();
+  };
+
+  const playAgainHandler = () => {
+    setHaveQuestions(false);
+    setCurrentQuestionIndex(0);
+    setQuizData({});
+    setIsFinished(false);
+    setScore(0);
+    setBtnDisabled(false);
+    setTimer(Date.now() + timerValue * 1000);
+    setTimerIndex((prevState) => prevState + 1);
+  };
 
   const countdownCompleteHandler = () => {
     if (currentQuestionIndex < questions.length - 1) {
-      setCurrentQuestionIndex(prevState => {
-        return prevState + 1
-      })
-
-      setTimer(Date.now() + time * 1000)
-      setTimerIndex(prevState => {
-        return prevState + 1
-      })
+      setCurrentQuestionIndex((prevState) => {
+        return prevState + 1;
+      });
+
+      setTimer(Date.now() + timerValue * 1000);
+      setTimerIndex((prevState) => {
+        return prevState + 1;
+      });
     } else {
-      setIsFinished(true)
+      setIsFinished(true);
+    }
+  };
+
+  const pauseTimer = () => countdownTimer.current.pause();
+
+  const resumeTimer = () => countdownTimer.current.start();
+
+  const updateStatistics = async () => {
+    // Verifica si user está definido y tiene la propiedad history
+    if (!user) {
+      console.error("User is undefined");
+      return;
     }
-  }
 
-  const pauseTimer = () => countdownTimer.current.pause()
-  console.log(questions)
+    console.log("User:", user);
+    if (!user.history) {
+      console.error("User history is undefined");
+      return;
+    }
+
+    // const passedQuestions = user.history.passedQuestions || 0;
+    // const failedQuestions = user.history.failedQuestions || 0;
+    // const gamesPlayed = user.history.gamesPlayed || 0;
+    // const timePlayed = user.history.timePlayed || 0;
+    // const points = user.history.points || 0;
+
+    const passedQuestions = 10;
+    const failedQuestions = 10;
+    const gamesPlayed = 10;
+    const timePlayed = 10;
+    const points = 10;
+
```

Comment:
It seems that the number of passed questions and other statistics are being harcoded, is this correct?